modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -165,6 +170,19 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
   ): F[Either[ResolutionError, SchemaList]] =
     listSchemasResult(vendor, name, model).map(_.map(_.value))
 
+  /**
+   * Vendor, name, model are extracted from supplied schema key to call on the `listSchemas`. The important difference
+   * from `listSchemas` is that it would invalidate cache, if returned list did not contain SchemaKey supplied in
+   * argument. Making it a safer option is latest schema bound is known.
+   */
+  def listSchemasLike(schemaKey: SchemaKey)(implicit
+    F: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): F[Either[ResolutionError, SchemaList]] =
```

Comment:
Is it possible to return here full `SchemaListLookupResult`? I suspect it would be useful for caching in transformer?

Reply:
This functions in the same the the `listSchemas` does. The same argument as before as consumers shouldn't need to know if it came from the cache or not. 

The transformer's result would be cached by the iglu-client, so I won't be building another caching wrapper.

Reply:
In general yes, caching details should be hidden. But we needed that to build related cache in the transfromer -> see : https://github.com/snowplow/iglu-scala-client/issues/207 and usage in transformer https://github.com/snowplow/snowplow-rdb-loader/commit/756b4a75187f24865ce6c6db894f53b7027f9c15

That's what I meant by 'caching in transformer' - on top of the cache in iglu-client build another cache for ddl `Properties` (in case of shredded type) and cache for ddl `Field` (in case of parquet type)

Reply:
What about:
>Is it possible to return here full SchemaListLookupResult

? When we add mentioned `Field` cache to transfromer, we'd need that. So maybe it's better to add full `SchemaListLookupResult` now and map on a consumer side. Otherwise, we would have to add this method in the next release or use full `listSchemasResult(schemaKey.vendor, schemaKey.name, schemaKey.version.model, Some(schemaKey))` which is a bit verbose?